gcs/src/core/charm_config.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""GCS charm configuration validation."""
+
+import logging
+import re
+from enum import StrEnum
+from typing import Optional
+
+from charms.data_platform_libs.v0.data_models import BaseConfigModel
+from pydantic import ConfigDict, Field, StrictStr, field_validator
+
+logger = logging.getLogger(__name__)
+
+
+class CharmConfigInvalidError(Exception):
+    """Configuration is invalid."""
+
+    def __init__(self, msg: str):
+        self.msg = msg
+        super().__init__(msg)
+
+
+class StorageClass(StrEnum):
+    """Type of storage classes supported by GCS."""
+
+    STANDARD = "STANDARD"
+    NEARLINE = "NEARLINE"
+    COLDLINE = "COLDLINE"
+    ARCHIVE = "ARCHIVE"
+
+
+# bucket: 3–63 chars, lowercase letters/digits/hyphens,
+# must start and end with a letter or digit
+_BUCKET_RX = re.compile(r"^[a-z0-9](?:[a-z0-9.-]{1,61})[a-z0-9]$")
+
+
+class CharmConfig(BaseConfigModel):
+    """Basic Config Validation.
+
+    Args:
+        bucket (StrictStr): Target GCS bucket. Syntax checks: (3–63, lowercase, digits, hyphens).
+        credentials (StrictStr): Juju Secret (id or label) that contains a service-account JSON (validated online).
+        storage_class (StrictStr): Optional storage class (STANDARD|NEARLINE|COLDLINE|ARCHIVE).
+        path (StrictStr): Optional object prefix <=1024 bytes UTF-8, no NULL, no leading slash (/).
+    """
+
+    model_config = ConfigDict(populate_by_name=True, extra="ignore")
+    bucket: StrictStr = Field(
+        ..., description="Target GCS bucket (3–63, lowercase/digits/hyphens)"
+    )
+    credentials: StrictStr = Field(
+        ..., description="Juju secret id/label holding service-account JSON"
+    )
+    storage_class: Optional[StrictStr] = Field(
+        default=None,
+        alias="storage-class",
+        description="GCS class (STANDARD|NEARLINE|COLDLINE|ARCHIVE)",
+    )
```

Comment:
You've added a default value in the config.yaml, can you please replicate it so there's no issue here ?
Otherwise there's a risk on inconsistency I think.